### PR TITLE
Add event person association

### LINKED ISSUE #105: Extend editEvent to allow for adding/removing of persons associated

### PR DESCRIPTION
**Pull request by LinkedInk**
Related classes has been added/updated

add -e command has been updated to be able to use ap/ (e.g. : add -e d/test at/12-12-1234 12:34 ap/1,2).
edit -e command has been updated to be able to use ap/ and rp/ (e.g. : edit -e 1 ap/1,2 rp/1,2).
ap/* and rp/* are accepted as add all and remove all respectively.
Removing is done first before adding.

Duplicate adding is not allowed.

Test cases are modified and updated to work with new constructor in AddEvent.

Implemented saving and reading of events with association.

### CODE REVIEW

**Comment by chan-j-d** [comment]
Hi, could you clarify the purpose of `FauxPerson` and why you chose to use it instead of a direct association with `Person`?

**Comment by LinkedInk** [comment]
> Hi, could you clarify the purpose of `FauxPerson` and why you chose to use it instead of a direct association with `Person`?

This is to simplify the `Person` class into something simple but still uniquely identifiable as details of the `Person` object being referenced is not important. Hence the `displayName` and `personhashCode`.

Storing of this class is simplified compared to storing entire copies of persons in `calendar.json` which would be needless.

Eventually (logic yet to be implemented), when editing the details of a contact, the logic will just have to compare and update the 2 fields, `displayName` and `personHashCode`. Cutting out unnecessary logic and processing time.  

On the filp side, getting the person details (logic yet to be implmented) means comparing the `personhashCode` to retrieve the actual `Person` to pull details from.

File: src/main/java/seedu/address/model/event/Event.java [hunk 1]
```diff
@@ -57,15 +76,26 @@ public boolean equals(Object other) {
...
     @Override
     public String toString() {
         final StringBuilder builder = new StringBuilder();
         builder.append(getDescription())
-                .append(" At: ")
+                .append("\nAt: ")
                 .append(getTime());
+
+        if (associatedPersons.size() != 0) {
+            builder.append("\n\nPeople attending: ");
```
Comment (solkim-83) [inline src/main/java/seedu/address/model/event/Event.java:91]: I believe this is a typo unless you mean 2 empty lines

**Review by solkim-83 — COMMENTED** [review]
the implementations and classes generally look good. 

it would be good to clarify the "typo". otherwise good to merge.

File: src/main/java/seedu/address/model/event/Event.java (hunk 1 above)
Comment (LinkedInk) [reply src/main/java/seedu/address/model/event/Event.java:91]: Its next line followed by skip 1 line. Example:
*info*
At: 12-12-1234 12:34(one \n here)
(one \n here)      
People attending: 
*info*

**Review by solkim-83 — APPROVED** [review]
lgtm!

**Pull request merged**